File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        s=0
        r=0
        flag =1
        if x<0:
            flag=-1
        k=str(abs(x))[::-1]
        ans= int(k)*flag
        if ans< -2**31 or ans> 2**31-1:
            return 0
        return ans
        

        

File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {

        long long r=0,s=0;
        bool flag=false;
        while(x!=0){
            r= x%10;
            x=x/10;
            if (s > INT_MAX / 10 || (s == INT_MAX / 10 && r > 7)) return 0;
            if (s < INT_MIN / 10 || (s == INT_MIN / 10 && r < -8)) return 0;
              s= s*10 +r;
        }

        return s;
    }
};